pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -840,24 +842,31 @@ func (m *kubeGenericRuntimeManager) computePodActions(ctx context.Context, pod *
 
 		// Get the containers to start, excluding the ones that succeeded if RestartPolicy is OnFailure.
 		var containersToStart []int
+		hasInitialized := false
 		for idx, c := range pod.Spec.Containers {
+			status := podStatus.FindContainerStatusByName(c.Name)
+			if status != nil && status.State != kubecontainer.ContainerStateUnknown {
+				// If there is a valid container status, it means the pod has been
+				// initialized.
+				hasInitialized = true
+			}
```

Comment:
comment to simplify the review:

calculate `hasInitialized` here and use it below (L860) as there is no `findNextInitContainerToRun` anymore. The logic is not changed.

In the worst case, all regular containers are `Unknown` state but then `len(containersToStart) > 0`.